### PR TITLE
feat(icons): remove Icon component and update docs

### PR DESCRIPTION
**Pull request by mnajdova**
# BREAKING CHANGES

- Removed Icon component, users should use some of the icons exported in `@fluentui/react-icons-northstar`
- Removed icons from the theme object
- Removed types: `TeamsSvgIconSpec`, `ThemeIconSpec`, `SvgIconSpec`, `SvgIconSpecWithStyles`

More details on https://github.com/microsoft/fluentui/pull/12489 for the migration steps.

***

This PR do some cleanups after https://github.com/microsoft/fluentui/pull/12489

- icons in the theme are removed
- `font-awesome` theme is removed
- `Icon` component is removed

### CODE REVIEW

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 1]
```diff
@@ -69,7 +80,7 @@ const ComponentControls: React.FC<ComponentControlsProps> = props => {
             aria-label={toolbarAriaLabel || null}
             items={[
               {
-                icon: <Icon name={'code'} {...{ style: { width: '20px', height: '20px' } }} />,
+                icon: <FilesCodeIcon {...{ style: { width: '20px', height: '20px' } }} />,
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:83]:
```suggestion
                icon: <FilesCodeIcon style={{ width: '20px', height: '20px' }} />,
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 2]
```diff
@@ -78,7 +89,7 @@ const ComponentControls: React.FC<ComponentControlsProps> = props => {
               },
 
               {
-                icon: <Icon {...{ name: 'paint brush', style: { width: '20px', height: '20px' } }} />,
+                icon: <FilesImageIcon {...{ style: { width: '20px', height: '20px' } }} />,
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:92]:
```suggestion
                icon: <FilesImageIcon style={{ width: '20px', height: '20px' }} />,
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 3]
```diff
@@ -91,15 +102,15 @@ const ComponentControls: React.FC<ComponentControlsProps> = props => {
                 kind: 'divider',
               },
               {
-                icon: <Icon name={'adjust'} {...{ style: { width: '20px', height: '20px' } }} />,
+                icon: <CircleIcon outline {...{ style: { width: '20px', height: '20px' } }} />,
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:105]:
```suggestion
                icon: <CircleIcon outline style={{ width: '20px', height: '20px' }} />,
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 4]
```diff
@@ -91,15 +102,15 @@ const ComponentControls: React.FC<ComponentControlsProps> = props => {
...
                 onClick: onShowTransparent,
                 active: showTransparent,
                 children: (Component, props) => (
                   <Tooltip content="Transparent" key="show-transparent" trigger={<Component {...props} />} />
                 ),
               },
               {
-                icon: <Icon {...{ name: 'align right', style: { width: '20px', height: '20px' } }} />,
+                icon: <IndentIcon rotate={180} {...{ style: { width: '20px', height: '20px' } }} />,
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:113]:
```suggestion
                icon: <IndentIcon rotate={180} style={{ width: '20px', height: '20px' }} />,
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 5]
```diff
@@ -108,7 +119,7 @@ const ComponentControls: React.FC<ComponentControlsProps> = props => {
               },
 
               {
-                icon: <Icon {...{ name: 'external alternate', style: { width: '20px', height: '20px' } }} />,
+                icon: <OpenOutsideIcon {...{ style: { width: '20px', height: '20px' } }} />,
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:122]:
```suggestion
                icon: <OpenOutsideIcon style={{ width: '20px', height: '20px' }} />,
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 6]
```diff
@@ -129,13 +140,13 @@ const ComponentControls: React.FC<ComponentControlsProps> = props => {
...
                 onClick: onCodeSandboxClick,
-                icon: <Icon {...{ name: codeSandboxIcon, style: { width: '20px', height: '20px' } }} />,
+                icon: codeSandboxIcon,
                 children: (Component, props) => (
                   <Tooltip content={codeSandboxTooltip} key="show-codesandbox" trigger={<Component {...props} />} />
                 ),
               },
               {
-                icon: <Icon name={'linkify'} {...{ style: { width: '20px', height: '20px' } }} />,
+                icon: <LinkIcon {...{ style: { width: '20px', height: '20px' } }} />,
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:149]:
```suggestion
                icon: <LinkIcon style={{ width: '20px', height: '20px' }} />,
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx [hunk 7]
```diff
@@ -1,6 +1,8 @@
 import * as React from 'react';
 import * as _ from 'lodash';
-import { Icon, Tooltip } from '@fluentui/react-northstar';
+import { Tooltip } from '@fluentui/react-northstar';
+// TODO: find replacement
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx:4]: Do we want to keep this todo?

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentDocLinks.tsx [hunk 8]
```diff
@@ -25,7 +25,7 @@ export default class ComponentDocLinks extends React.PureComponent<any, any> {
         }}
       >
         <>
-          <Icon name="github" />
+          <Image src="public/images/github.png" width="16px" height="16px" />
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentDocLinks.tsx:28]: This looks like a good case to create a custom icon with the factory 🤔

Comment (mnajdova) [reply packages/fluentui/docs/src/components/ComponentDoc/ComponentDocLinks.tsx:28]: Not sure if I can just copy svg from other libraries.. If you can think of some source let me know..

File: packages/fluentui/react-northstar/src/themes/teams/types.ts [hunk 9]
```diff
@@ -107,7 +107,7 @@ export type TeamsThemeStylesProps = {
   Grid: GridProps;
   Header: HeaderProps;
   HeaderDescription: HeaderDescriptionProps;
-  Icon: IconStylesProps;
+  Icon: SvgIconStylesProps;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/types.ts:110]: This looks weird. Should it be `SvgIcon`?

```suggestion
  SvgIcon: SvgIconStylesProps;
```

**Comment by layershifter** [comment]
This file `packages/fluentui/react-northstar/src/themes/teams/withProcessedIcons.ts` should be also removed

**Comment by layershifter** [comment]
Some types were removed `TeamsSvgIconSpec`, `ThemeIconSpec`, `SvgIconSpec`, `SvgIconSpecWithStyles`. Let's add this to breaking changes

File: packages/fluentui/react-northstar/src/themes/teams/components/SvgIcon/svgIconVariables.ts [hunk 10]
```diff
@@ -30,7 +30,7 @@ export interface IconVariables {
   sizeModifier?: IconSizeModifier;
 }
 
-export default (siteVars): Partial<IconVariables> => ({
+export default (siteVars): Partial<SvgIconVariables> => ({
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/SvgIcon/svgIconVariables.ts:33]:
```suggestion
export default (siteVars): SvgIconVariables => ({
```

Should `Partial` be there?

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx (hunk 7 above)
Comment (mnajdova) [reply packages/fluentui/docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx:4]: Nope

File: packages/fluentui/react-northstar/src/themes/teams/components/SvgIcon/svgIconVariables.ts (hunk 10 above)
Comment (mnajdova) [reply packages/fluentui/react-northstar/src/themes/teams/components/SvgIcon/svgIconVariables.ts]: No, good call, it was IconVariables previosly

File: packages/fluentui/react-northstar/src/components/SvgIcon/SvgIcon.tsx [hunk 11]
```diff
@@ -13,6 +13,11 @@ import { ThemeContext } from 'react-fela';
 
 import { ProviderContextPrepared, WithAsProp, withSafeTypeForAs } from '../../types';
 
+export type SvgIconStylesProps = Pick<
+  SvgIconProps,
+  'bordered' | 'circular' | 'disabled' | 'outline' | 'rotate' | 'size' | 'xSpacing'
+>;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/SvgIcon/SvgIcon.tsx:19]:
```suggestion
export type SvgIconStylesProps = Required<Pick<
  SvgIconProps,
  'bordered' | 'circular' | 'disabled' | 'outline' | 'rotate' | 'size' | 'xSpacing'
>>;
```

And let's also use in `useStyles` generic: `useStyles<SvgIconStylesProps>`

File: packages/fluentui/docs/src/views/Theming.tsx [hunk 12]
```diff
@@ -99,13 +99,13 @@ export default () => (
     <ExampleSnippet
       render={() => (
         <>
-          <Icon name="calendar" circular />
-          <Icon name="calendar" circular />
+          <CalendarIcon circular />
+          <CalendarIcon circular />
 
           <Provider theme={{ componentVariables: { Icon: { color: 'cornflowerblue' } } }}>
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/Theming.tsx:105]:
```suggestion
          <Provider theme={{ componentVariables: { SvgIcon: { color: 'cornflowerblue' } } }}>
```

??

File: packages/fluentui/docs/src/prototypes/chatPane/composeMessage.tsx [hunk 13]
```diff
@@ -89,7 +89,7 @@ const getMenuItems = (): ShorthandCollection<MenuItemProps> => {
     key: `${index}-${name}`,
     icon: {
       name,
```
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/chatPane/composeMessage.tsx:91]: Oops 💣

File: packages/fluentui/docs/src/prototypes/chatPane/chatPaneHeader.tsx [hunk 14]
```diff
@@ -84,20 +81,21 @@ class ChatPaneHeader extends React.PureComponent<ChatPaneHeaderProps> {
...
-              margin: 'auto',
-              ...(!index && { margin: 'auto 1.6rem auto auto' }),
-            }}
-            variables={siteVars => ({ color: siteVars.colors.grey[350] })}
-          />
-        ))}
+        {[TeamCreateIcon, MoreIcon].map((IconComponent, index) => {
+          return (
+            <IconComponent
+              key={`${index}-${name}`}
```
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/chatPane/chatPaneHeader.tsx:87]: `name` doesn't exist in this scope

```suggestion
              key={index}
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentDocLinks.tsx (hunk 8 above)
Comment (layershifter) [reply packages/fluentui/docs/src/components/ComponentDoc/ComponentDocLinks.tsx:28]: Probably it's not safe due licensing issues 🤔

**Pull request merged**